src/mpid/ch4/shm/ipc/gpu/gpu_post.h

```diff
@@ -36,19 +36,15 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_GPU_get_mem_attr(const void *vaddr, MPIDI_IPC
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_GPU_GET_MEM_ATTR);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_IPC_GET_MEM_ATTR);
 
-#ifdef MPIDI_CH4_SHM_ENABLE_GPU
-    attr->ipc_type = MPIDI_IPCI_TYPE__GPU;
-    if (MPIR_CVAR_MPL_ENABLE_GPU) {
+    if (MPIDI_CH4_SHM_ENABLE_GPU /* shmmod control at configure */  &&
+        MPIR_CVAR_MPL_ENABLE_GPU /* overall control at runtime */) {
+        attr->ipc_type = MPIDI_IPCI_TYPE__GPU;
         MPL_gpu_ipc_get_mem_handle(&attr->mem_handle.gpu.ipc_handle, (void *) vaddr);
         attr->threshold.send_lmt_sz = MPIR_CVAR_CH4_GPU_LMT_MSG_SIZE;
     } else {
-        memset(&attr->mem_handle.gpu.ipc_handle, 0, sizeof(MPL_gpu_ipc_mem_handle_t));
+        attr->ipc_type = MPIDI_IPCI_TYPE__NONE;
         attr->threshold.send_lmt_sz = MPIR_AINT_MAX;
     }
-#else
-    attr->ipc_type = MPIDI_IPCI_TYPE__NONE;
-    attr->threshold.send_lmt_sz = MPIR_AINT_MAX;
-#endif
```

Comment:
This entire piece can simply be replaced with:

```
    attr->ipc_type = MPIDI_IPCI_TYPE__GPU;
    MPL_gpu_ipc_get_mem_handle(&attr->mem_handle.gpu.ipc_handle, (void *) vaddr);
    attr->threshold.send_lmt_sz = MPIR_CVAR_CH4_GPU_LMT_MSG_SIZE;
```

The reason is that we'd never reach this code, if the user disables GPU support at runtime through the MPL CVAR.